webapp/src/components/rhs/rhs.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useState, useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import styled from 'styled-components';
+
+import {makeGetPostsInChannel, getPost} from 'mattermost-redux/selectors/entities/posts';
+import {getPosts} from 'mattermost-redux/actions/posts';
+
+import {manifest} from '@/manifest';
+
+import ThreadItem from './thread_item';
+import RHSHeader from './rhs_header';
+import RHSNewTab from './rhs_new_tab';
+
+const ThreadViewer = styled((window as any).Components.ThreadViewer)`
+    height: 100%;
+`;
+
+const ThreadsList = styled.div`
+    overflow-y: scroll;
+`;
+
+const RhsContainer = styled.div`
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+`;
+
+export default function RHS() {
+    const dispatch = useDispatch();
+    const [currentTab, setCurrentTab] = useState('new');
+
+    const botChannelId = useSelector((state: any) => state['plugins-' + manifest.id].botChannelId);
+    const getPostsInChannel = makeGetPostsInChannel();
+    let posts = useSelector((state) => getPostsInChannel(state as any, botChannelId || '', -1)) || [];
+    posts = posts.sort((a, b) => b.update_at - a.update_at);
+
+    const selectedPostId = useSelector((state: any) => state['plugins-' + manifest.id].selectedPostId);
```

Comment:
There doesn't seem to be a need for this to go though redux. Maybe `useState` would be fine here?

Reply:
The reason for this is because in different places we can select an AI post (using the `selectPost` action), that is not easy accessible from that places, like the post menu. I think this requires to be at redux or any other kind of global state to be able to be modified from other places.